Engine/Client/Src/Systems/DrawOBJ/DrawOBJSystem.cpp

```diff
@@ -20,11 +24,27 @@ void DrawOBJSystem::_drawOBJ(ECS::Registry& reg, int idxPacketEntities)
     std::shared_ptr<IGraphic> libGraphic = getGraphicalLibrary();
 
     ECS::SparseArray<IComponent> objPathComponents = reg.get_components<IComponent>("ObjPathComponent");
+    ECS::SparseArray<IComponent> position3DComponents = reg.get_components<IComponent>("Position3DComponent");
+    ECS::SparseArray<IComponent> scaleComponents = reg.get_components<IComponent>("ScaleComponent");
 
-    for (auto entity : reg.getEntities()) {
+    for (ECS::entity_t entity = 0; objPathComponents.size() >= entity + 1; entity++) {
         std::shared_ptr<ObjPathComponent> obj = std::dynamic_pointer_cast<ObjPathComponent>(objPathComponents[entity]);
-        if (obj)
-            libGraphic->drawOBJ(obj->path, entity * 2, 0, 0, 1);
+
+        std::shared_ptr<Position3DComponent> pos = (position3DComponents.size() >= entity + 1) ?
+            std::dynamic_pointer_cast<Position3DComponent>(position3DComponents[entity]) : nullptr;
```

Comment:
What does this line (same for the scale) do ?

Reply:
This lines check if an entity has a position and a scale or not, because they are optional.